src/Samples/NewWindow.Core/Window2.fs

```diff
@@ -28,37 +28,38 @@ type Window2OutMsg =
 
 module Window2 =
   module Input =
-    let get m = m.Input
-    let set v m = { m with Input = v }
+    let get (m: Window2) : string = m.Input
+    let set (v: string) (m: Window2) : Window2 = { m with Input = v }
   module IsChecked =
-    let get m = m.IsChecked
-    let set v m = { m with IsChecked = v }
+    let get (m: Window2) : bool = m.IsChecked
+    let set (v: bool) (m:Window2) : Window2 = { m with IsChecked = v }
   module ConfirmState =
-    let set v m = { m with ConfirmState = v }
+    let set (v: ConfirmState option) (m: Window2) : Window2 = { m with ConfirmState = v }
 
-  let init =
+  let init : Window2 =
     { Input = ""
       IsChecked = false
       ConfirmState = None }
 
-  let update = function
+  let update (msg: Window2Msg) : (Window2 -> Window2) =
+    match msg with
     | SetInput s -> s |> Input.set
     | SetChecked b -> b |> IsChecked.set
     | Submit -> ConfirmState.Submit |> Some |> ConfirmState.set
     | Cancel -> ConfirmState.Cancel |> Some |> ConfirmState.set
     | Close  -> ConfirmState.Close  |> Some |> ConfirmState.set
 
-  let private confirmStateVisibilityBinding confirmState =
+  let private confirmStateVisibilityBinding (confirmState: ConfirmState) : (string -> Binding<Window2,'a>) =
     fun m -> m.ConfirmState = Some confirmState
     >> Bool.toVisibilityCollapsed
     |> Binding.oneWay
 
-  let private confirmStateToMsg confirmState msg m =
+  let private confirmStateToMsg (confirmState: ConfirmState) (msg: 'a) (m: Window2) : InOut<'a,Window2OutMsg> =
     if m.ConfirmState = Some confirmState
     then InOut.Out Window2OutMsg.Close
     else InOut.In msg
 
-  let bindings () =
+  let bindings unit : Binding<Window2,InOut<Window2Msg,Window2OutMsg>,obj> list =
```

Comment:
I strongly object to naming `unit` parameters. They are meant to appear similar to parameterless function calls in other languages; if we name them, we lose the resemblance.

Reply:
@LyndonGingerich I get your objection but offer that an exception might be made for consistent parameterization of the `Samples`.